cdist/conf/type/__letsencrypt_cert/gencode-remote

```diff
@@ -16,6 +16,12 @@ case "${state}" in
 	;;
 	present)
 		requested_domains="${__object}/parameter/domain"
+        tmp_domains=$(mktemp domain.cdist.XXXXXXXXXX)
+        if [ -f "${requested_domains}" ]; then
+            cp "${requested_domains}" "${tmp_domains}"
+        else
+            echo "$__object_id" >> "${tmp_domains}"
+        fi
```

Comment:
Why don't you just write `${__object_id}` to `${requested_domains}` if the file does not exist? I believe using a temporary file is overkill. Also, the script uses tabs instead of spaces for indentation.

Reply:
But we want to have original parameter values in .cdist/cache, don't we?

Reply:
@telmich What would you prefer here, to temp or not to temp? :)

Reply:
@darko-poljak Not sure if I made myself clear. So, I was wrong at first and I think that you're right, a temporary file should be used. What I don't like is naming of the variable that holds path to the temporary file. I think that `${requested_domains}` is more suitable than `${tmp_domains}` for the temporary file and the current `${requested_domains}` should be renamed to something else, or perhaps just get removed (it's used only twice).

Reply:
@lubo Aaa, I misunderstood you, sorry.

Reply:
@lubo Renamed.